src/main/java/com/ita/if103java/ims/dao/impl/WarehouseDaoImpl.java

```diff
@@ -42,9 +43,19 @@ private PreparedStatement createWarehouseStatement(Warehouse warehouse, Connecti
         statement.setString(++i, warehouse.getInfo());
         statement.setInt(++i, warehouse.getCapacity());
         statement.setBoolean(++i, warehouse.isBottom());
-        statement.setLong(++i, warehouse.getParentID());
+        if (warehouse.getParentID() == null) {
+            statement.setNull(++i, Type.LONG);
+        } else {
+            statement.setLong(++i, warehouse.getParentID());
+        }
```

Comment:
@volodymyrvynnyk , I saw that in the EventDaoImpl you store transaction id and warehouse id as setObject. Does this handle null values?

If yes, then @Mychalevytsh can use 
statement.setObject(++i, warehouse.getParentID());
statement.setObject(++i, warehouse.getTopWarehouseId());
instead of if-else constructions and not known imports

Reply:
@ikrasWork , yes, it can store object and null either